Chess/Assets/Scripts/ArrayChess/PieceMover.cs

```diff
@@ -54,83 +49,54 @@ public class PieceMover : MonoBehaviour
     private Vector3[] chessSquaresDistance;
 
     private float targetDistance = 1f;
-    private float loadtimer = 1f;
 
     void Start()
     {
         rend = GetComponent<Renderer>();
         pieceStartColor = rend.material.color;
-
+        chessSquares = GameObject.FindGameObjectsWithTag("square"); // Populates the chessboard array of all square positions 
```

Comment:
This one line is all we need to replace the `finalAllSquares` function! It was an unecessary abstration, I think we did that a few times :P